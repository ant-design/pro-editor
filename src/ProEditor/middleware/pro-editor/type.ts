import { StateCreator } from 'zustand/vanilla';
import { ConfigPublicAction } from '../../store/slices/config';
import { TakeTwo, Write } from '../types/utils';

export interface ProEditorSetStateAction {
  type: unknown;

  /**
   * 如果记录，那么历史记录的名字是什么
   */
  name?: string;
  /**
   * 是否将操作记录到历史记录中
   */
  recordHistory?: boolean;
}

/**
 * 用于给注入方法添加额外的 proEditor 实例对象
 */
export interface ProEditorMiddlewareInjectMethod<Sa extends unknown[], Sr, T, A> {
  setState<A1 extends string | ProEditorSetStateAction>(...a: [...a: TakeTwo<Sa>, action?: A1]): Sr;

  proEditor: {
    undo?: () => void;
    redo?: () => void;
    // clearStorage: () => void;
    getOptions: () => Partial<ProEditorOptions<T, A>>;
  };
}
/**
 * 提供给用户的配置项
 */
export interface ProEditorOptions<S, EditorSaveState = S> {
  /** Name of the storage (must be unique) */
  name: string;
  /**
   * Filter the persisted value.
   *
   * @params state The state's value
   */
  partialize?: (state: S) => EditorSaveState;
}

// 为  mutator 注入 'pro-editor' 类型，以支持第三个配置参数

type WithProEditor<S, A> = S extends {
  getState: () => infer T;
  setState: (...a: infer Sa) => infer Sr;
}
  ? Write<S, ProEditorMiddlewareInjectMethod<Sa, Sr, T, A>>
  : never;

declare module 'zustand/vanilla' {
  interface StoreMutators<S, A> {
    ['pro-editor']: WithProEditor<S, A>;
  }
}

// 内部方法，用于保证中间件内部的类型定义安全

export type ProEditorImpl = <T>(
  storeInitializer: StateCreator<T, [['pro-editor', unknown]], []>,
  options: ProEditorOptions<T, T>,
) => StateCreator<T, [['pro-editor', unknown]], []>;

export interface InjectInternalProEditor {
  proEditor: {
    __INTERNAL_SET_CONFIG__NOT_USE_IT: ConfigPublicAction['setConfig'];
  };
}
