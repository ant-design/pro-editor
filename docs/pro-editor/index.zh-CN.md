---
title: ProBuilder 配置容器
atomId: ProBuilder
group: 组件
nav:
  title: 装配器
  order: 3
---

# ProBuilder 装配器

使用 `ProBuilder` 可以快速构建出组件的配置界面。

## 代码演示

<code src='./demos/buttonAssets.tsx' compact="true" title="Button 示例"></code>

<code src='./demos/defaultAssets.tsx' title="传入 `ComponentAssets`"></code>

<code src='./demos/empty.tsx' title="无 `ComponentAssets`"></code>

<code src="./demos/controlledPresence.tsx" title="视图状态受控" description="将 `editorAwareness` 设为受控状态，可以控制当前的视窗状态。"></code>

## API

<API id="ProEditor"></API>
