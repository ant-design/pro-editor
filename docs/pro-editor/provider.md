---
title: ProEditorProvider
group: 组件
---

# ProEditorProvider

如果：

- 希望在不是 ProEditor 组件子元素的组件中使用 `useProEditor`、useConfig、`useViewport` 等 ProEditor 钩子；
- 在页面上有多个 ProEditor 实例；
- 需要在不是 ProEditor 组件子元素的组件中访问内部状态，或使用 ProEditor 所提供的数据方法。

那么你就需要用 `ProEditorProvider` 包裹相应的应用组件。
