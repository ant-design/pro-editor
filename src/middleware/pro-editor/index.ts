// import {  } from '@/ProEditor/store';
import { StateCreator, StoreMutatorIdentifier } from 'zustand/vanilla';
import { ProEditorImpl, ProEditorSetStateAction } from './type';

/**
 * 提供给用户的配置项
 */
export interface ProEditorOptions<S, EditorSaveState = S> {
  /** Name of the storage (must be unique) */
  name: string;
  /**
   * Filter the persisted value.
   *
   * @params state The state's value
   */
  partialize?: (state: S) => EditorSaveState;
}

const middleware: ProEditorImpl = (storeInitializer, options) => (set, get, api) => {
  const partialize = options.partialize ?? ((s) => s);
  /**
   * 记录历史
   * @param action
   */
  const updateInProEditor = (action: ProEditorSetStateAction) => {
    const nextConfig = partialize(get());

    get().proEditor.__INTERNAL_SET_CONFIG__NOT_USE_IT(nextConfig, {
      trigger: 'proEditorMiddleware',
      ...action,
    });
  };
  /*
   * Capture the initial state so that we can initialize the pro editor store to the
   * same values as the initial values of the Zustand store.
   */
  const store = storeInitializer(
    /*
     * Create a new set function that defers to the original and then passes
     * the new state to patchSharedType.
     */
    (partial, replace, action) => {
      set(partial, replace, action);
      updateInProEditor((action as any) || {});
    },
    get,
    {
      ...api,
      // Create a new setState function as we did with set.
      setState: (partial, replace, action) => {
        api.setState(partial, replace, action);

        updateInProEditor((action as any) || {});
      },
    },
  );

  // Return the initial state to create or the next middleware.
  return {
    ...store,
    proEditor: { options: { ...options, partialize } },
  };
};

export type ProEditorMiddleware = <
  T,
  Mps extends [StoreMutatorIdentifier, unknown][] = [],
  Mcs extends [StoreMutatorIdentifier, unknown][] = [],
  U = T,
>(
  initializer: StateCreator<T, [...Mps, ['pro-editor', unknown]], Mcs>,
  options: ProEditorOptions<T, U>,
) => StateCreator<T, Mps, [['pro-editor', U], ...Mcs]>;

export const proEditorMiddleware = middleware as unknown as ProEditorMiddleware;
