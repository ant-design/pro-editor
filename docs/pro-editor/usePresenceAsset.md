---
title: usePresenceAsset
group: Hooks
order: 2
---

该方法可以获取编辑器中的资产的感知视图状态

```ts
import { usePresenceAsset } from '@ant-design/pro-editor';

const getPresenceAsset = () => {
  const [presence, setPresence] = usePresenceAsset();

  useEffect(() => {
    console.log(setPresence);
  }, [presence]);

  return null;
};
```

:::warning

注：该 hook 只能在被 `ProEditorProvider` 包裹的组件中使用，或者在 `<ProEditor />` 组件的子组件中使用。

:::
