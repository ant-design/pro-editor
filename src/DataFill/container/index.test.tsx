import { render } from '@testing-library/react';

import { DataFill } from '@ant-design/pro-editor';

describe('DataFill', () => {
  it('数据填充可正常渲染', () => {
    const { container } = render(<DataFill />);
    expect(container).toMatchSnapshot();
  });

  // 阻塞执行，暂时注释
  //   it('带有点击事件', async () => {
  //     // 设置 onChange 方法
  //     const keyMap = [];
  //     let fillfunc;
  //     const onClick = (payload: FilledPayload) => {
  //       keyMap.push(payload.key);
  //       fillfunc = payload.filler;
  //     };
  //     const { findByText } = render(<DataFill onClick={onClick} />);

  //     const nameDom = await findByText('文本');

  //     await act(async () => {
  //       await userEvent.click(nameDom);
  //     });
  //     expect(keyMap).toEqual(['common']);
  //     expect(fillfunc()).toBeDefined();
  //   });
});
