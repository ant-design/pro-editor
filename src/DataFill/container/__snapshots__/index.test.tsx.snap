// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DataFill > 数据填充可正常渲染 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-1 .ant-tree-directory .ant-tree-switcher,
.emotion-1 .ant-tree-directory .ant-tree-iconEle {
  line-height: 32px!important;
}

.emotion-1 .ant-tree-directory .ant-tree-node-content-wrapper {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding-left: 0;
  line-height: 32px!important;
}

.emotion-1 .ant-tree-directory .ant-tree-title {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-1 .ant-tree-directory .ant-tree-treenode {
  cursor: pointer;
}

.emotion-1 .ant-tree-directory .ant-tree-treenode::before {
  border: 1px solid rgba(5, 5, 5, 0.06);
  border-radius: 4px;
}

.emotion-1 .ant-tree-directory .ant-tree-treenode:hover .ant-tree-switcher {
  color: rgba(0, 0, 0, 0.88);
}

.emotion-1 .ant-tree-directory .ant-tree-treenode:hover .ant-tree-node-content-wrapper {
  color: rgba(0, 0, 0, 0.88);
}

.emotion-1 .ant-tree-directory .ant-tree-treenode:hover::before {
  background: rgba(0, 0, 0, 0.02);
  border-color: #1677ff;
  box-shadow: 0 0 3px 0 #1677ff;
}

.emotion-1 .ant-tree-directory .ant-tree-treenode .ant-tree-switcher-noop {
  cursor: pointer;
}

.emotion-1 .ant-tree-directory .ant-tree-treenode .ant-tree-switcher {
  color: rgba(0, 0, 0, 0.88);
}

.emotion-1 .ant-tree-directory .ant-tree-treenode .ant-tree-node-content-wrapper {
  color: rgba(0, 0, 0, 0.88);
}

.emotion-1 .ant-tree-directory .ant-tree-treenode-selected::before {
  background: #ffffff;
  border: 2 solid #1677ff;
}

.emotion-1 .ant-tree-directory .ant-tree-treenode-selected .ant-tree-switcher {
  color: rgba(0, 0, 0, 0.88);
}

.emotion-1 .ant-tree-directory .ant-tree-treenode-selected .ant-tree-node-content-wrapper {
  color: rgba(0, 0, 0, 0.88);
}

.emotion-1 .ant-tree-directory .ant-tree-indent-unit {
  width: 12px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  width: 100%;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 8px;
}

.emotion-5 {
  max-width: 120px;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  color: rgba(0, 0, 0, 0.25);
  font-size: 12px;
}

.emotion-6 {
  padding: 4px;
}

.emotion-6 .studio-segmented-item {
  margin-right: 4px;
  color: rgba(0, 0, 0, 0.45);
  border-radius: 2px;
}

.emotion-6 .studio-segmented-item:hover {
  color: rgba(0, 0, 0, 0.65);
}

.emotion-6 .studio-segmented-item:last-child {
  margin-right: 0;
}

.emotion-6 .studio-segmented-item-selected {
  color: rgba(0, 0, 0, 0.88);
}

.emotion-6 .studio-segmented-item-selected:hover {
  color: rgba(0, 0, 0, 0.88);
}

.emotion-6 .studio-segmented-item-label {
  min-height: auto;
  padding: 2px 7px;
  line-height: unset;
}

<div>
  <div
    class="layoutkit-flexbox emotion-0 emotion-1"
  >
    <div
      class="ant-tree ant-tree-block-node ant-tree-unselectable ant-tree-directory"
      role="tree"
    >
      <div>
        <input
          aria-label="for screen reader"
          style="width: 0px; height: 0px; display: flex; overflow: hidden; opacity: 0; border: 0px; padding: 0px; margin: 0px;"
          tabindex="0"
          value=""
        />
      </div>
      <div
        aria-hidden="true"
        class="ant-tree-treenode"
        style="position: absolute; pointer-events: none; visibility: hidden; height: 0px; overflow: hidden; border: 0px; padding: 0px;"
      >
        <div
          class="ant-tree-indent"
        >
          <div
            class="ant-tree-indent-unit"
          />
        </div>
      </div>
      <div
        class="ant-tree-list"
        style="position: relative;"
      >
        <div
          class="ant-tree-list-holder"
        >
          <div>
            <div
              class="ant-tree-list-holder-inner"
              style="display: flex; flex-direction: column;"
            >
              <div
                aria-grabbed="false"
                class="ant-tree-treenode ant-tree-treenode-switcher-close"
                draggable="false"
              >
                <span
                  aria-hidden="true"
                  class="ant-tree-indent"
                />
                <span
                  class="ant-tree-switcher ant-tree-switcher_close"
                >
                  <span
                    aria-label="caret-down"
                    class="anticon anticon-caret-down ant-tree-switcher-icon"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="caret-down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="0 0 1024 1024"
                      width="1em"
                    >
                      <path
                        d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                      />
                    </svg>
                  </span>
                </span>
                <span
                  class="ant-tree-node-content-wrapper ant-tree-node-content-wrapper-close"
                  title="文本"
                >
                  <span
                    class="ant-tree-iconEle ant-tree-icon__customize"
                  >
                    <svg
                      class="pro-editor-icon"
                      fill="currentColor"
                      height="1em"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 1024 1024"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M853.3 1024H170.7A171.2 171.2 0 0 1 0 853.3V170.7C0 76.8 76.8 0 170.7 0h682.6C947.2 0 1024 76.8 1024 170.7v682.6c0 93.9-76.8 170.7-170.7 170.7zM170.7 85.3c-47 0-85.4 38.4-85.4 85.4v682.6c0 47 38.4 85.4 85.4 85.4h682.6c47 0 85.4-38.4 85.4-85.4V170.7c0-47-38.4-85.4-85.4-85.4H170.7z"
                      />
                      <path
                        d="M725.3 341.3H298.7c-25.6 0-42.7-17-42.7-42.6s17-42.7 42.7-42.7h426.6c25.6 0 42.7 17 42.7 42.7s-17 42.6-42.7 42.6z"
                      />
                      <path
                        d="M298.7 384c-25.6 0-42.7-17-42.7-42.7v-42.6c0-25.6 17-42.7 42.7-42.7s42.6 17 42.6 42.7v42.6c0 25.6-17 42.7-42.6 42.7zM725.3 384c-25.6 0-42.6-17-42.6-42.7v-42.6c0-25.6 17-42.7 42.6-42.7s42.7 17 42.7 42.7v42.6c0 25.6-17 42.7-42.7 42.7zM512 768c-25.6 0-42.7-17-42.7-42.7V298.7c0-25.6 17.1-42.7 42.7-42.7s42.7 17 42.7 42.7v426.6c0 25.6-17.1 42.7-42.7 42.7z"
                      />
                      <path
                        d="M554.7 768h-85.4c-25.6 0-42.6-17-42.6-42.7s17-42.6 42.6-42.6h85.4c25.6 0 42.6 17 42.6 42.6s-17 42.7-42.6 42.7z"
                      />
                    </svg>
                  </span>
                  <span
                    class="ant-tree-title"
                  >
                    <div
                      class="layoutkit-flexbox emotion-2"
                    >
                      <div
                        class="layoutkit-flexbox emotion-3"
                      >
                        <div
                          class="layoutkit-flexbox emotion-0"
                          style="white-space: nowrap;"
                        >
                          文本
                        </div>
                        <span
                          class="ant-editor-demo-data emotion-5"
                        />
                      </div>
                      <div
                        style="white-space: nowrap;"
                      >
                        <div
                          class="ant-segmented emotion-6 ant-segmented-sm"
                          style="line-height: 1;"
                        >
                          <div
                            class="ant-segmented-group"
                          >
                            <label
                              class="ant-segmented-item ant-segmented-item-selected"
                            >
                              <input
                                checked=""
                                class="ant-segmented-item-input"
                                type="radio"
                              />
                              <div
                                class="ant-segmented-item-label"
                                title="短文本"
                              >
                                短文本
                              </div>
                            </label>
                            <label
                              class="ant-segmented-item"
                            >
                              <input
                                class="ant-segmented-item-input"
                                type="radio"
                              />
                              <div
                                class="ant-segmented-item-label"
                                title="长文本"
                              >
                                长文本
                              </div>
                            </label>
                          </div>
                        </div>
                      </div>
                    </div>
                  </span>
                </span>
              </div>
              <div
                aria-grabbed="false"
                class="ant-tree-treenode ant-tree-treenode-switcher-close"
                draggable="false"
              >
                <span
                  aria-hidden="true"
                  class="ant-tree-indent"
                />
                <span
                  class="ant-tree-switcher ant-tree-switcher_close"
                >
                  <span
                    aria-label="caret-down"
                    class="anticon anticon-caret-down ant-tree-switcher-icon"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="caret-down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="0 0 1024 1024"
                      width="1em"
                    >
                      <path
                        d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                      />
                    </svg>
                  </span>
                </span>
                <span
                  class="ant-tree-node-content-wrapper ant-tree-node-content-wrapper-close"
                  title="编号"
                >
                  <span
                    class="ant-tree-iconEle ant-tree-icon__customize"
                  >
                    <span
                      aria-label="folder"
                      class="anticon anticon-folder"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="folder"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M880 298.4H521L403.7 186.2a8.15 8.15 0 00-5.5-2.2H144c-17.7 0-32 14.3-32 32v592c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V330.4c0-17.7-14.3-32-32-32zM840 768H184V256h188.5l119.6 114.4H840V768z"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    class="ant-tree-title"
                  >
                    <div
                      class="layoutkit-flexbox emotion-2"
                    >
                      <div
                        class="layoutkit-flexbox emotion-3"
                      >
                        <div
                          class="layoutkit-flexbox emotion-0"
                          style="white-space: nowrap;"
                        >
                          编号
                        </div>
                        <span
                          class="ant-editor-demo-data emotion-5"
                        />
                      </div>
                    </div>
                  </span>
                </span>
              </div>
              <div
                aria-grabbed="false"
                class="ant-tree-treenode ant-tree-treenode-switcher-close"
                draggable="false"
              >
                <span
                  aria-hidden="true"
                  class="ant-tree-indent"
                />
                <span
                  class="ant-tree-switcher ant-tree-switcher_close"
                >
                  <span
                    aria-label="caret-down"
                    class="anticon anticon-caret-down ant-tree-switcher-icon"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="caret-down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="0 0 1024 1024"
                      width="1em"
                    >
                      <path
                        d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                      />
                    </svg>
                  </span>
                </span>
                <span
                  class="ant-tree-node-content-wrapper ant-tree-node-content-wrapper-close"
                  title="数据"
                >
                  <span
                    class="ant-tree-iconEle ant-tree-icon__customize"
                  >
                    <svg
                      class="pro-editor-icon"
                      height="1em"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 1195 897"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                      xmlns:xlink="http://www.w3.org/1999/xlink"
                    >
                      <g
                        fill="none"
                        fill-rule="evenodd"
                        id="FillNumber-页面-1"
                        stroke="none"
                        stroke-width="1"
                      >
                        <g
                          fill="currentColor"
                          fill-rule="nonzero"
                          id="FillNumber-icon-number"
                          transform="translate(0.207180, 0.557698)"
                        >
                          <path
                            d="M74.666667,74.666667 L74.666667,821.333333 L1120,821.333333 L1120,74.666667 L74.666667,74.666667 Z M74.666667,-1.42108547e-14 L1120,-1.42108547e-14 C1161.216,-1.42108547e-14 1194.66667,33.450667 1194.66667,74.666667 L1194.66667,821.333333 C1194.66667,862.570595 1161.23726,896 1120,896 L74.666667,896 C33.4294055,896 0,862.570595 0,821.333333 L0,74.666667 C0,33.450667 33.450667,-1.42108547e-14 74.666667,-1.42108547e-14 Z M263.68,263.68 L314.965333,263.68 L314.965333,680.106667 L246.698667,680.106667 L246.698667,346.026667 C219.34797,370.032502 186.811608,387.381334 151.637333,396.714667 L151.637333,329.045333 C172.473579,323.511105 192.49732,315.277848 211.2,304.554667 C232.192,291.754667 249.6,278.357334 263.68,263.765333 L263.68,263.68 Z M548.949333,255.488 C589.141333,255.488 622.336,267.178667 648.704,290.474667 C674.304,313.856 687.104,343.552 687.104,380.928 C687.104,417.109333 673.194667,449.792 646.314667,479.488 C630.016,496.981333 600.832,519.168 560.042667,546.56 C517.376,574.549333 491.776,599.125333 482.389333,620.117333 L687.786667,620.117333 L687.786667,680.192 L397.226667,680.192 C397.226667,637.525333 410.624,600.832 438.698667,569.258667 C453.802667,551.850667 485.888,526.165334 534.272,492.885333 C561.152,474.282666 579.84,458.496 591.445333,446.208 C609.621333,425.813333 618.922666,403.626667 618.922667,380.330667 C618.922667,357.546667 612.522667,340.650667 600.832,329.557333 C588.544,318.464 570.538667,313.258666 546.56,313.258667 C520.96,313.258667 501.674667,321.962667 488.874667,339.541333 C476.074667,355.754666 468.992,380.928 467.797333,413.610667 L399.530667,413.610667 C400.128,366.933334 413.610667,329.557334 440.490667,300.970667 C467.797334,270.677334 503.978667,255.488 548.864,255.488 L548.949333,255.488 Z M896.597333,255.488 C938.581333,255.488 972.970666,265.386667 998.656,285.866667 C1023.744,306.261334 1036.544,334.250667 1036.544,370.346667 C1036.544,415.914667 1013.248,446.293334 967.168,461.397333 C991.658667,469.077333 1010.944,480.085333 1023.744,495.274667 C1037.73867,510.976 1044.736,531.456 1044.736,555.861333 C1044.736,594.432 1031.33867,625.92 1004.45867,650.410667 C976.469334,675.498667 939.776,688.298667 894.208,688.298667 C851.114667,688.298667 816.042667,677.205334 789.845333,655.018667 C760.661333,630.613334 744.362666,594.432 740.864,547.754667 L810.24,547.754667 C811.434667,574.549334 819.626667,595.541334 835.84,610.133333 C850.517333,623.616 869.717333,630.613333 893.696,630.613333 C919.893333,630.613333 940.885333,622.933333 956.074667,608.426667 C969.474285,595.450839 976.890408,577.497642 976.554667,558.848 C976.554667,534.954667 968.874667,517.376 954.88,506.368 C940.885333,494.677333 920.490667,489.386667 893.696,489.386667 L864.512,489.386667 L864.512,438.101333 L893.696,438.101333 C918.186667,438.101333 936.874667,432.810666 949.674667,422.314667 C961.877334,411.818667 968.362667,396.117334 968.362667,375.637333 C968.362667,355.242666 962.474667,340.053333 951.381333,329.557333 C939.178666,319.061333 921.088,313.856 897.194667,313.856 C872.704,313.856 854.016,319.658667 840.618667,331.946667 C826.624,344.149334 818.432,362.837334 816.042667,387.925333 L748.970667,387.925333 C752.469334,345.941333 767.744,313.258666 795.648,289.877333 C821.930667,266.581333 855.722667,255.488 896.597333,255.488 L896.597333,255.488 Z"
                            id="FillNumber-形状"
                          />
                        </g>
                      </g>
                    </svg>
                  </span>
                  <span
                    class="ant-tree-title"
                  >
                    <div
                      class="layoutkit-flexbox emotion-2"
                    >
                      <div
                        class="layoutkit-flexbox emotion-3"
                      >
                        <div
                          class="layoutkit-flexbox emotion-0"
                          style="white-space: nowrap;"
                        >
                          数据
                        </div>
                        <span
                          class="ant-editor-demo-data emotion-5"
                        />
                      </div>
                      <div
                        style="white-space: nowrap;"
                      >
                        <div
                          class="ant-segmented emotion-6 ant-segmented-sm"
                          style="line-height: 1;"
                        >
                          <div
                            class="ant-segmented-group"
                          >
                            <label
                              class="ant-segmented-item"
                            >
                              <input
                                class="ant-segmented-item-input"
                                type="radio"
                              />
                              <div
                                class="ant-segmented-item-label"
                                title="0~9"
                              >
                                0~9
                              </div>
                            </label>
                            <label
                              class="ant-segmented-item"
                            >
                              <input
                                class="ant-segmented-item-input"
                                type="radio"
                              />
                              <div
                                class="ant-segmented-item-label"
                                title="10~99"
                              >
                                10~99
                              </div>
                            </label>
                            <label
                              class="ant-segmented-item ant-segmented-item-selected"
                            >
                              <input
                                checked=""
                                class="ant-segmented-item-input"
                                type="radio"
                              />
                              <div
                                class="ant-segmented-item-label"
                                title="100~999"
                              >
                                100~999
                              </div>
                            </label>
                          </div>
                        </div>
                      </div>
                    </div>
                  </span>
                </span>
              </div>
              <div
                aria-grabbed="false"
                class="ant-tree-treenode ant-tree-treenode-switcher-close"
                draggable="false"
              >
                <span
                  aria-hidden="true"
                  class="ant-tree-indent"
                />
                <span
                  class="ant-tree-switcher ant-tree-switcher_close"
                >
                  <span
                    aria-label="caret-down"
                    class="anticon anticon-caret-down ant-tree-switcher-icon"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="caret-down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="0 0 1024 1024"
                      width="1em"
                    >
                      <path
                        d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                      />
                    </svg>
                  </span>
                </span>
                <span
                  class="ant-tree-node-content-wrapper ant-tree-node-content-wrapper-close"
                  title="时间"
                >
                  <span
                    class="ant-tree-iconEle ant-tree-icon__customize"
                  >
                    <span
                      aria-label="clock-circle"
                      class="anticon anticon-clock-circle"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="clock-circle"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                        />
                        <path
                          d="M686.7 638.6L544.1 535.5V288c0-4.4-3.6-8-8-8H488c-4.4 0-8 3.6-8 8v275.4c0 2.6 1.2 5 3.3 6.5l165.4 120.6c3.6 2.6 8.6 1.8 11.2-1.7l28.6-39c2.6-3.7 1.8-8.7-1.8-11.2z"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    class="ant-tree-title"
                  >
                    <div
                      class="layoutkit-flexbox emotion-2"
                    >
                      <div
                        class="layoutkit-flexbox emotion-3"
                      >
                        <div
                          class="layoutkit-flexbox emotion-0"
                          style="white-space: nowrap;"
                        >
                          时间
                        </div>
                        <span
                          class="ant-editor-demo-data emotion-5"
                        />
                      </div>
                      <div
                        style="white-space: nowrap;"
                      >
                        <div
                          class="ant-segmented emotion-6 ant-segmented-sm"
                          style="line-height: 1;"
                        >
                          <div
                            class="ant-segmented-group"
                          >
                            <label
                              class="ant-segmented-item ant-segmented-item-selected"
                            >
                              <input
                                checked=""
                                class="ant-segmented-item-input"
                                type="radio"
                              />
                              <div
                                class="ant-segmented-item-label"
                                title="日期时间"
                              >
                                日期时间
                              </div>
                            </label>
                            <label
                              class="ant-segmented-item"
                            >
                              <input
                                class="ant-segmented-item-input"
                                type="radio"
                              />
                              <div
                                class="ant-segmented-item-label"
                                title="日期"
                              >
                                日期
                              </div>
                            </label>
                            <label
                              class="ant-segmented-item"
                            >
                              <input
                                class="ant-segmented-item-input"
                                type="radio"
                              />
                              <div
                                class="ant-segmented-item-label"
                                title="时间"
                              >
                                时间
                              </div>
                            </label>
                          </div>
                        </div>
                      </div>
                    </div>
                  </span>
                </span>
              </div>
              <div
                aria-grabbed="false"
                class="ant-tree-treenode ant-tree-treenode-switcher-close ant-tree-treenode-leaf-last"
                draggable="false"
              >
                <span
                  aria-hidden="true"
                  class="ant-tree-indent"
                />
                <span
                  class="ant-tree-switcher ant-tree-switcher_close"
                >
                  <span
                    aria-label="caret-down"
                    class="anticon anticon-caret-down ant-tree-switcher-icon"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="caret-down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="0 0 1024 1024"
                      width="1em"
                    >
                      <path
                        d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                      />
                    </svg>
                  </span>
                </span>
                <span
                  class="ant-tree-node-content-wrapper ant-tree-node-content-wrapper-close"
                  title="地址"
                >
                  <span
                    class="ant-tree-iconEle ant-tree-icon__customize"
                  >
                    <span
                      aria-label="environment"
                      class="anticon anticon-environment"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="environment"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M854.6 289.1a362.49 362.49 0 00-79.9-115.7 370.83 370.83 0 00-118.2-77.8C610.7 76.6 562.1 67 512 67c-50.1 0-98.7 9.6-144.5 28.5-44.3 18.3-84 44.5-118.2 77.8A363.6 363.6 0 00169.4 289c-19.5 45-29.4 92.8-29.4 142 0 70.6 16.9 140.9 50.1 208.7 26.7 54.5 64 107.6 111 158.1 80.3 86.2 164.5 138.9 188.4 153a43.9 43.9 0 0022.4 6.1c7.8 0 15.5-2 22.4-6.1 23.9-14.1 108.1-66.8 188.4-153 47-50.4 84.3-103.6 111-158.1C867.1 572 884 501.8 884 431.1c0-49.2-9.9-97-29.4-142zM512 880.2c-65.9-41.9-300-207.8-300-449.1 0-77.9 31.1-151.1 87.6-206.3C356.3 169.5 431.7 139 512 139s155.7 30.5 212.4 85.9C780.9 280 812 353.2 812 431.1c0 241.3-234.1 407.2-300 449.1zm0-617.2c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm79.2 255.2A111.6 111.6 0 01512 551c-29.9 0-58-11.7-79.2-32.8A111.6 111.6 0 01400 439c0-29.9 11.7-58 32.8-79.2C454 338.6 482.1 327 512 327c29.9 0 58 11.6 79.2 32.8C612.4 381 624 409.1 624 439c0 29.9-11.6 58-32.8 79.2z"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    class="ant-tree-title"
                  >
                    <div
                      class="layoutkit-flexbox emotion-2"
                    >
                      <div
                        class="layoutkit-flexbox emotion-3"
                      >
                        <div
                          class="layoutkit-flexbox emotion-0"
                          style="white-space: nowrap;"
                        >
                          地址
                        </div>
                        <span
                          class="ant-editor-demo-data emotion-5"
                        />
                      </div>
                    </div>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
